core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -412,7 +412,6 @@ object CachedDataFrame
     val buffer = new Array[Byte](4 << 10) // 4K
     // final output is written to this buffer
     val output = new ByteBufferDataOutput(4 << 10,
-      DirectBufferAllocator.instance(),
```

Comment:
JVM sets the default limit on DirectBuffer allocations as per free RAM at startup. I guess the issue might be that it could be too small in some cases? But users can increase it explicitly in any case for pure heap case using JVM standard system property. I think that's better than hitting a genuine OOME due to lack of accounting. We can keep the HeapAllocator accounting for a bit later given that we expect enterprise deployments to all use off-heap where this is not an issue.